### PR TITLE
fix: grouped legend bug

### PR DESCRIPTION
**Pull request by itssharmasandeep**

### CODE REVIEW

**Review by palbizu — APPROVED** [review]
LGTM

File: projects/observability/src/shared/components/cartesian/d3/legend/cartesian-legend.ts [hunk 1]
```diff
@@ -41,7 +41,8 @@ export class CartesianLegend<TData> {
     private readonly summaries: Summary[] = []
   ) {
     this.isGrouped =
-      this.series.length > 0 && this.series.every(seriesEntry => seriesEntry.groupName !== seriesEntry.name);
+      this.series.length > 0 &&
+      this.series.every(seriesEntry => !isEmpty(seriesEntry.groupName) && seriesEntry.groupName !== seriesEntry.name);
```
Comment (aaron-steinfeld) [inline projects/observability/src/shared/components/cartesian/d3/legend/cartesian-legend.ts:45]: we're getting empty group name and series names?

Comment (itssharmasandeep) [reply projects/observability/src/shared/components/cartesian/d3/legend/cartesian-legend.ts:45]: I will take a look more into it, it is having a problem in app-dev

Comment (itssharmasandeep) [reply projects/observability/src/shared/components/cartesian/d3/legend/cartesian-legend.ts:45]: in events page

Comment (aaron-steinfeld) [reply projects/observability/src/shared/components/cartesian/d3/legend/cartesian-legend.ts:45]: Ah crap, that's in staging too, we're going to need to rush an upgrade. How'd we miss this, this is the only spot this feature is used!

**Pull request merged**